packages/notifi-frontend-client/lib/NotifiFrontendClient.ts

```diff
@@ -13,6 +13,16 @@ import { notNullOrEmpty } from './utils/notNullOrEmpty';
 
 export type SignMessageFunction = (message: Uint8Array) => Promise<Uint8Array>;
 
+type LoginProps = {
+  walletPublicKey: string;
+  readonly signMessage: SignMessageFunction;
+};
+
+type SignMessageProps = {
+  walletPublicKey: string;
+  readonly signMessage: SignMessageFunction;
+  readonly timestamp: number;
+};
```

Comment:
why can't these be fully Readonly?

```
type LoginProps = Readonly<{
  ...
}>
```

Reply:
Good point here, yes, walletAddress should not be changes as well